syntax = "proto3";

package ray;

message ForwardTaskRequest {
  // The ID of the task to be forwarded.
  bytes task_id = 1;
  // The tasks in the uncommitted lineage of the forwarded task. This
  // should include task_id.
  repeated bytes uncommitted_tasks = 2;
}

message ForwardTaskReply {}

service NodeManagerService {
  rpc ForwardTask(ForwardTaskRequest) returns (ForwardTaskReply);
}
